cms/static/js/views/modals/course_outline_modals.js

```diff
@@ -822,16 +830,33 @@ define(['jquery', 'backbone', 'underscore', 'gettext', 'js/views/baseview',
             return this.$('#staff_lock').is(':checked');
         },
 
+        setHideFromTOC: function(value) {
+            this.$('#hide_from_toc').prop('checked', value);
+        },
+
+        setVisibleToLearners: function(value) {
+            this.$('#visible_to_learners').prop('checked', value);
+        },
+
+        isVisibleToLearners: function() {
+            return this.$('#staff_lock').is(':not(:checked)') && this.$('#hide_from_toc').is(':not(:checked)');
+        },
+
+        isHiddenFromTOC: function() {
+            return this.$('#hide_from_toc').is(':checked');
+        },
+
         hasChanges: function() {
-            return this.isModelLocked() !== this.isLocked();
+            return this.isModelLocked() !== this.isLocked() || this.isModelHiddenFromTOC() !== this.isHiddenFromTOC();
         },
 
         getRequestData: function() {
             if (this.hasChanges()) {
                 return {
                     publish: 'republish',
                     metadata: {
-                        visible_to_staff_only: this.isLocked() ? true : null
+                        visible_to_staff_only: this.isLocked() ? true : null,
+                        hide_from_toc: this.isHiddenFromTOC() ? true : null
```

Comment:
```suggestion
                        visible_to_staff_only: this.isLocked() || null,
                        hide_from_toc: this.isHiddenFromTOC() || null
```

Reply:
@mariajgrimaldi Only this comment for my side

Reply:
Thanks for the suggestion. Although I wouldn't say I like introducing changes unrelated to the implementation itself, this definitely looks better. Thanks!